### PR TITLE
Document -Wdelete-non-virtual-dtor

### PR DESCRIPTION
**Pull request by ktf**

### CODE REVIEW

File: c++warnings.markdown [hunk 1]
```diff
@@ -31,3 +31,25 @@ See also:
 
 - <http://en.cppreference.com/w/cpp/memory/new/operator_new>
 - <http://en.cppreference.com/w/cpp/memory/new/operator_delete>
+
+## error: delete called on 'BaseClass' that has virtual functions but non-virtual destructor [-Werror,-Wdelete-non-virtual-dtor]
+
+Whenever a C++ class has virtual methods and it's therefore meant to
+be used polymorphically, it's good practice to make sure to provide a
+virtual destructor:
```
Comment (sawenzel) [inline c++warnings.markdown:39]: good practice might actually sound too weak?

Comment (ktf) [inline c++warnings.markdown:39]: True, but given that I added `-Werror=delete-non-virtual-dtor` in AliPhysics, people will not be able to compile their code in any case.

**Comment by sawenzel** [comment]
The merge has conflicts. Can you resolve them before we go on?

**Comment by ktf** [comment]
I did a mistake and pushed it directly. Sorry about that. Closing this.

**Pull request closed without merging**